src/canisters/frontend/ic-certified-assets/src/asset_certification/types/http.rs

```diff
@@ -114,14 +114,22 @@ impl HttpRequest {
         #[cfg(test)]
         let canister_id = self.get_canister_id();
 
-        let location = format!("https://{canister_id}.icp0.io{path}", path = self.url);
+        let location = match self.get_header_value("Host") {
+            Some(host_header) if host_header.contains("ic0.app") => {
```

Comment:
See https://dfinity.atlassian.net/browse/FOLLOW-983 for similar issue
Can we do this?
```suggestion
            Some(host_header) if host_header.ends_with("ic0.app") => {
```